packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -134,30 +156,20 @@ enum SdkIntKindsEnum {
   uint64 = "uint64",
 }
 
-enum SdkFloatKindsEnum {
+enum SdkFloatingPointKindsEnum {
   float = "float",
   float32 = "float32",
   float64 = "float64",
+}
+
+enum SdkFixedPointKindsEnum {
   decimal = "decimal",
   decimal128 = "decimal128",
 }
 
-const SdkAzureBuiltInStringKindsMapping = {
-  uuid: "uuid",
```

Comment:
May not relevant to this pr but I notice uuid is azure scope concept but url is not, i was wondering if uuid also is a general concept and should be in our bulit-in type in compiler?

Reply:
now if this `isStdType` function from compiler returns `true` for one scalar, it will be a builtin type kind.
Namely things defined [here](https://github.com/microsoft/typespec/blob/222fd806a018faff841837ac740c10c25222ef97/packages/compiler/lib/intrinsics.tsp#L9) and [here](https://github.com/microsoft/typespec/blob/222fd806a018faff841837ac740c10c25222ef97/packages/compiler/lib/std/types.tsp#L8)

I have similar opinion that uuid might be general enough to be in the standard library. But unfortunately it is not right now.
Please feel free to open issues in typespec repo for this.
And I have added some mechanism that once their kind for std type has new variant, we will know from the TCGC side (get compilation errors)